int svtype has conflict with perl CORE

Tux opened the pull request.

nrdvana commented:
Thanks!  How about the variable name 'obj_svtype' though, since the type doesn't relate to xlib?  And feel free to add yourself to CONTRIBUTORS.

Tux commented:
1. Renamed `xlib_svtype` to `obj_svtype`
2. Added myself to `CONTRIBUTORS` on your request
3. Did a spell-check + corrections. See [`pod-spell-check`](https://github.com/Tux/Release-Checklist/blob/master/scripts/pod-spell-check) in [Release::Checklist](https://github.com/Tux/Release-Checklist)
4. Update `cpanfile based` on known CVE's (consider promoting that to Makefile.PL)
5. Fixed `PREOP` in `Makefile.PL`

Enjoy

nrdvana commented on lib/X11/Xlib/Display.pm:
I assume this was needed for coverage tests?  Seems like it might render a bit redundantly.  Is it possible to put the L<> links as the list tag itself, without a content block?  Otherwise I'd rather just add a pod comment to mark them as covered.

nrdvana commented on lib/X11/Xlib/Struct.pm:
```diff
@@ -137,6 +137,12 @@ Access the scalar holding the bytes of the struct.
+C<buffer> is an alias tu L</C<bytes>>
+
+=cut
+
```
I'm pretty sure what I was doing with the aliases was renaming a part of the API and providing backward-compatibility for any code that used 'buffer'.  I don't really want to advertise the old names anymore.  Same with 'id' vs. 'xid' below.

nrdvana commented:
Thanks for all the updates!  I don't mind updating the version requirements in the makefile as long as it isn't breaking for old perls.  (X11 itself is a bit 'legacy', so I figured I'd try to keep 5.8 working)

Not sure why spell check hates contractions so much, but I guess that's OK.

I never realized the versions were supposed to update automatically on 'make dist' :-)  I always just run it by hand.

I know some modules don't declare strict or version, but that's because there's literally no code in them, aside from a line to load the file that actually declares them.  I guess it wouldn't *hurt* anything to re-declare the version though.

Tux commented:
> 

My Text::CSV_XS works back to 5.6.1, so I know the pain. Devel::PPPort and `perlver` are your friends. Hence the new tests in `xt/`

What version updates? Mine were by had too. If you mean the two META files, there are generated by `make dist` unless you provide them yoursel, in which case you should put them in MANIFEST and in git

If the module really has no code at all, I think that warrants a bug report, for it feels like a false positive. OTOH, a `use strict;` would not hurt either

nrdvana commented:
> What version updates?

When you fixed `dist => { PREOP => 'perl util/update_versions.pl' },` that causes the versions on all other files to get updated to match the version in lib/X11/Xlib.pm when I run `make dist`.

BTW, is there a Makefile target to run the xt/ tests?  Or do you just use options on `prove`?

Tux commented:
I altered the `PREOP` entry because on of my tests told me that `PREOP` was not a recognized top-level entry. Then I went on reading the ExtUtils::MakeMaker docs to find it was only documented as entry inside the top-level `dist` entry. As `update_versions.pl` is your own and not part of MakeMaker, I think you can do all checks and modifications inside that script.

There are probably a zillion ways to include `xt/` in your tests. `xt/` is meant as "developer-only" tests, so that should be guarded. I do it like [this](https://github.com/Tux/Data-Peek/blob/master/Makefile.PL#L97)
